index.js

```diff
@@ -132,65 +132,82 @@ const getSecurityVulnerabilitiesForBitbucketRepo = async (
 ) => {
   console.log(`Looking for vulnerabilities in Bitbucket ${repo}`)
   let allVulnerabilities
-  
-  const javaScriptDependencies = await bitbucket.getJavaScriptDependenciesOfRepo(
-    bitbucketUsername,
-    bitbucketAppPassword,
-    repo
-  )
-  if (javaScriptDependencies === undefined) {
-    console.log('(No JavaScript dependencies found)')
-  } else {
-    const javaScriptVulnerabilities = await getSecurityVulnerabilitiesForJavaScriptDependencies(
-      githubToken,
-      javaScriptDependencies
-    )
-    const javaScriptVulnerabilitiesWithRepo = javaScriptVulnerabilities.map(vulnerability => addRepoName(repo, vulnerability))
-    allVulnerabilities = allVulnerabilities || []
-    allVulnerabilities.push(...javaScriptVulnerabilitiesWithRepo)
-  }
-  
-  const phpDependencies = await bitbucket.getPhpDependenciesOfRepo(
-    bitbucketUsername,
-    bitbucketAppPassword,
-    repo
-  )
-  if (phpDependencies === undefined) {
-    console.log('(No PHP dependencies found)')
-  } else {
-    const phpVulnerabilities = await getSecurityVulnerabilitiesForPhpDependencies(
-      githubToken,
-      phpDependencies
-    )
-    const phpVulnerabilitiesWithRepo = phpVulnerabilities.map(vulnerability => addRepoName(repo, vulnerability))
-    allVulnerabilities = allVulnerabilities || []
-    allVulnerabilities.push(...phpVulnerabilitiesWithRepo)
-  }
-  
-  if (versionsCsvUrl) {
-    const nodeJsVersionVulnerabilities = await getNodeJsVersionVulnerabilitiesOfBitbucketRepo(
+
+  try {
+    const javaScriptDependencies = await bitbucket.getJavaScriptDependenciesOfRepo(
```

Comment:
Could/should this have been accomplished by extracting most of the code inside the `try` into a separate function?